contracts/pair/src/contract.rs

```diff
@@ -157,10 +172,37 @@ impl LiquidityPoolTrait for LiquidityPool {
             &"POOL".into_val(&env),
         );
 
+        let stake_contract_address = utils::deploy_stake_contract(&env, stake_wasm_hash);
+        stake_contract::Client::new(&env, &stake_contract_address).initialize(
+            &admin,
+            &share_token_address,
+            &min_bond,
+            &max_distributions,
+            &min_reward,
+        );
+
+        let stake_args = (
+            // admin
+            &admin,
+            // lp_token
+            &share_token_address,
+            min_bond,
+            max_distributions,
+            min_reward,
+        )
+            .into_val(&env);
+
+        env.invoke_contract::<Val>(
+            &stake_contract_address,
+            &Symbol::new(&env, "initialize"),
+            stake_args,
+        );
```

Comment:
You already initialize staking contract. Why then passing arguments second time and invoking?
```suggestion
        stake_contract::Client::new(&env, &stake_contract_address).initialize(
            &admin,
            &share_token_address,
            &min_bond,
            &max_distributions,
            &min_reward,
        );
```

Reply:
I hate myself for this